docs/src/main/asciidoc/qute.adoc

```diff
@@ -0,0 +1,271 @@
+----
+<1> `{name}` is a value expression that is evaluated when the template is rendered.
+
+NOTE: By default, all files located in the `META-INF/resources/templates` directory and its subdirectories are registered as templates. Templates are validated during startup and watched for changes in the development mode.
+
+Now let's inject the "compiled" template in the resource class.
+
+.HelloResource.java
+[source,java]
+----
+package org.acme.quarkus.sample;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+import io.quarkus.qute.Rendering;
+import io.quarkus.qute.Template;
+
+@Path("hello")
+public class HelloResource {
+
+    @Inject
+    Template hello; <1>
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public Rendering get(@QueryParam("name") String name) {
+        return hello.render().data("name", name); <2> <3>
+    }
+}
+----
+<1> If there is no `@ResourcePath` qualifier provided the field name is used to locate the template. In this particular case, we're injecting a template with path `META-INF/resources/templates/hello.txt`.
+<2> `Template.render()` returns a new rendering action that can be customized before the actual rendering is triggered. In this case, we put the name value under the key `name`. The data map is accessible during rendering. 
+<3> Note that we don't trigger the rendering - this is done automatically by a special `ContainerResponseFilter` implementation.
+
+If running your application, you can request the endpoint:
+
+```
+$ curl -w "\n" http://localhost:8080/hello?name=Martin
+Hello Martin!
+```
+
+== Parameter Declarations and Template Extension Methods
+
+Qute has many useful features.
+In this example, we'll demonstrate two of them.
+If you declare a *parameter declaration* in a template then Qute attempts to validate all expressions that reference this parameter and if an incorrect expression is found the build fails.
+*Template extension methods* are used to extend the set of accessible properties of data objects.
+
+Let's suppose we have a simple class like this:
+
+.Item.java
+[source,java]
+----
+public class Item {
+    public String name;
+    public BigDecimal price;
+}
+----
+
+And we'd like to render a simple HTML page that contains the item name, price and also a discounted price.
+The discounted price is sometimes called a "computed property".
+We will implement a template extension method to render this property easily.
+Let's start again with the template:
+
+.item.html
+[source,html]
+----
+{@org.acme.Item item} <1>
+<!DOCTYPE html>
+<html>
+<head>
+<meta charset="UTF-8">
+<title>{item.name}</title> <2>
+</head>
+<body>
+    <h1>{item.name}</h1>
+    <div>Price: {item.price}</div> 
+    {#if item.price > 100} <3>
+    <div>Discounted Price: {item.discountedPrice}</div> <4>
+    {/if}
+</body>
+</html>
+----
+<1> Optional parameter declaration. Qute attempts to validate all expressions that reference paramater `item`.
```

Comment:
How multiple parameters declarations are handled (should they be declared in another line or separated by a comma)?

Reply:
Each param declaration is in a new tag. Anyway, this will be covered in the ref guide I'm currently working on. There's so many things and the guide should be brief and concise.